### DISCUSSION TITLE
kubernetes-handbook/objects/pod-hook/

### DISCUSSION THREAD

**Discussion by giscus** [rootsongjc/cloud-native-library#96] [Announcements]
# kubernetes-handbook/objects/pod-hook/

Pod Hook（钩子）是由 Kubernetes 管理的 kubelet 发起的，当容器中的进程启动前或者容器中的进程终止之前运行，这是包含在容器的生命周期之中。可以同时为 Pod 中的所有容器都配置 hook。 Hook 的类型包括两种： exec：执行一段命令

https://lib.jimmysong.io/kubernetes-handbook/objects/pod-hook/

**Comment by hl8086** [comment]
"该操作跟容器的启动命令是在同步执行的，也就是说在 postStart 操作执行完成之前，kubelet 会锁住容器，不让应用程序的进程启动"--------请问这里是不是有问题啊，还是我理解错了？官方文档--“postStart 处理函数与容器的代码是异步执行的”，而且我自己验证的时候在postStart里面执行sleep，然后进去容器发现业务进程已经启动了

**Reply by rootsongjc** [reply]
官方文档上确实是说的 preStart 和容器中的端点代码是 **异步** 执行的，而且在 preStart 执行完成之前 Pod 的状态都不会是 RUNNING。但是这也仅仅是影响 Pod 的状态而已，并不是说在 preStart 执行完成之前组织容器端点的代码执行。

**Comment by shaofuzhang** [comment]
大佬。preStop章节是在更新吗？

**Reply by rootsongjc** [reply]
这个在两个月前已经更新过了。
